### PR TITLE
If 425 transform assets table to primeng

### PR DESCRIPTION
**Pull request by florian-paul**

### CODE REVIEW

**Comment by mattmikulina** [comment]
Task linked: [IF-425 Transform table assets of FactoryOwner to primeNG table](https://app.clickup.com/t/2177764/IF-425)

File: fusionfrontend/src/app/components/ui/table-filter/table-filter.component.ts [hunk 1]
```diff
@@ -109,7 +109,7 @@ export class TableFilterComponent implements OnInit {
   }
 
   addFilter() {
-    this.filters.every(filter => {
+    this.tableFilters.every(filter => {
```
Comment (jsylupp) [inline fusionfrontend/src/app/components/ui/table-filter/table-filter.component.ts:112]: is every the wanted method here or forEach?

File: fusionfrontend/src/assets/sass/components/_primetable.scss [hunk 2]
```diff
@@ -165,4 +165,78 @@
...
+    align-self: center;
+    margin-right: 48px;
+  }
+}
+
+
+.table-dropdown-menu-button {
+  outline: none;
+  background-color: transparent !important;
+  color: #95989a !important;
```
Comment (jsylupp) [inline fusionfrontend/src/assets/sass/components/_primetable.scss:201]: use variables

File: fusionfrontend/src/assets/sass/components/_primetable.scss [hunk 3]
```diff
@@ -165,4 +165,78 @@
...
+    height: 40px !important;
+    font-size: 14px !important;
+    font-family: Metropolis, "Avenir Next", Helvetica, serif;
+    padding-left: 2em !important;
+    padding-right: 2em !important;
+    white-space: nowrap !important;
+  }
+
+   .p-menuitem-icon {
+    font-size: 16px;
```
Comment (jsylupp) [inline fusionfrontend/src/assets/sass/components/_primetable.scss:228]: use variable

File: fusionfrontend/src/assets/sass/components/_primetable.scss [hunk 4]
```diff
@@ -165,4 +165,78 @@
...
+    margin-top: 0.25em !important;
+  }
+
+   .p-menu-overlay {
+    top: unset !important;
+  }
+
+   .p-menuitem-link {
+    height: 40px !important;
+    font-size: 14px !important;
```
Comment (jsylupp) [inline fusionfrontend/src/assets/sass/components/_primetable.scss:220]: use variable for font-size and extract values of font-family to new variable and add it on all places

File: fusionfrontend/src/app/factory/components/pages/factory-site-page/factory-site-page.component.ts [hunk 5]
```diff
@@ -82,17 +82,8 @@ export class FactorySitePageComponent implements OnInit, OnDestroy {
...
-  }
-
-  updateAssetData(event: FactoryAssetDetails) {
-    event.id = event.id ? event.id : this.createdAssetDetailsId;
-    event.companyId = this.companyId;
-    this.assetService.updateCompanyAsset(this.companyId, event).subscribe(
-      () => { },
-      error => console.error(error)
-    );
+  selectTheAssets(selectedAssetIds: ID[]) {
```
Comment (jsylupp) [inline fusionfrontend/src/app/factory/components/pages/factory-site-page/factory-site-page.component.ts:85]: (selectAssets would also be sufficient)

File: fusionfrontend/src/app/factory/components/content/assets-list/assets-list.component.ts [hunk 6]
```diff
@@ -143,57 +176,108 @@ export class AssetsListComponent implements OnInit {
...
+  deleteAsset() {
+    this.assetService.removeCompanyAsset(this.activeListItem.companyId, this.activeListItem.id).subscribe(() => {
+      this.factoryAssetDetailsWithFields.splice(this.factoryAssetDetailsWithFields.indexOf(this.activeListItem), 1);
+    });
   }
 
-  onAssetDeselect(asset: FactoryAssetDetailsWithFields) {
-    this.selectedIds.delete(asset.id);
-    this.emitSelectEvent();
+  searchAssets(event?): void {
```
Comment (jsylupp) [inline fusionfrontend/src/app/factory/components/content/assets-list/assets-list.component.ts:191]: maybe add type of event, also below

File: fusionfrontend/src/app/factory/components/content/assets-list/assets-list.component.html [hunk 7]
```diff
@@ -13,51 +13,93 @@
...
+    <fa-icon [icon]="faCheckCircle" class="check-box-icon fa-fw"></fa-icon>
+    <span class="table-selected-item">{{ selectedFactoryAssets.length | i18nPlural: editBarMapping }}</span>
+  </div>
+
+  <div class="p-col-4 table-selected-item" style="text-align: center">
+    <div class="table-bar-button-div">
+      <p-button class="table-bar-button" (click)="onCardsViewClick()">
+        <fa-icon [icon]="faThLarge" class="check-box-icon fa-fw"></fa-icon>
+      </p-button>
+      <p-button *ngIf="selectedFactoryAssets.length == 1" class="table-bar-button" (click)="setActiveRow(); showEditDialog();">
```
Comment (jsylupp) [inline fusionfrontend/src/app/factory/components/content/assets-list/assets-list.component.html:41]: use ===, also below

File: fusionfrontend/src/app/factory/components/content/assets-list/assets-list.component.html [hunk 8]
```diff
@@ -13,51 +13,93 @@
...
 </div>
+
+<p-table [(selection)]="selectedFactoryAssets" [paginator]="true" [rowsPerPageOptions]="[5,10,50]" [rows]="10"
+         [showCurrentPageReport]="true" [value]="displayedFactoryAssets" [rowHover]="true"
+         class="list-container" currentPageReportTemplate="{first} to {last} of {totalRecords} Assets">
+  <ng-template pTemplate="header">
+    <tr>
+      <th [style.width.%]="5"></th>
+      <th [style.width.%]="30">
+        <app-table-searchbar filterColumn="Asset" [itemsToBeFiltered]="this.factoryAssetDetailsWithFields"
```
Comment (jsylupp) [inline fusionfrontend/src/app/factory/components/content/assets-list/assets-list.component.html:65]: maybe remove this

**Review by jsylupp — CHANGES_REQUESTED** [review]
Well done :) just some little suggestions for change

**Pull request merged**